src/iterator.rs

```diff
@@ -154,13 +149,38 @@ pub trait PointCloud: Sync {
     type Id: ToString + Send + Copy;
     fn nodes_in_location(&self, location: &PointLocation) -> Vec<Self::Id>;
     fn encoding_for_node(&self, id: Self::Id) -> Encoding;
-    fn points_in_node<'a>(
-        &'a self,
-        query: &'a PointQuery,
+    /// Return all points in the selected node.
+    fn points_in_node(
+        &self,
+        attributes: &[&str],
         node_id: Self::Id,
         batch_size: usize,
-    ) -> Result<FilteredIterator<'a>>;
+    ) -> Result<NodeIterator>;
     fn bounding_box(&self) -> &Aabb3<f64>;
+
+    /// Return the points matching the query in the selected node.
+    /// Why only a single node? Because the nodes are distributed to several `PointStream`s working
```

Comment:
i think you meant `Pointstreams`

Reply:
now i get it...

Reply:
maybe  Pointstream instances would be nicer to read thoug

Reply:
It was kind of intentional – to highlight the name of the struct, without the plural. Do you prefer to change it?

Reply:
Changed to PointStream instances